tests/scripts/all.sh

```diff
@@ -1246,6 +1246,14 @@ component_test_default_cmake_gcc_asan_new_bignum () {
     tests/context-info.sh
 }
 
+component_test_crypto_with_static_key_slots() {
```

Comment:
If keystore without malloc is a feature, we should make a build where only libtestdriver1 and test code has access to malloc, but the `mbedtls_calloc` used to build the core always fails. (Of course no md/cipher/pk/x509/tls in that build.)

Reply:
I think I need a bit of help on this. The [last commit](https://github.com/Mbed-TLS/mbedtls/pull/9302/commits/b38c62fefd40d7584bf6aa0b9e2f9d538ddb2c8f) I added tries to implement what (I think) you asked in this comment, i.e. a test component where `libtestdriver1` and `test_suite_xxx` can use `malloc/free` whereas the library (including PSA core) cannot. First question: albeit this is a rough implementation is that close to what you had in mind?

Reply:
This looks on the right track: compiling the core with different settings for malloc/free compared with libtestdriver1 and the test code. It's incomplete because with `MBEDTLS_PLATFORM_MEMORY` undefined, `mbedtls_calloc` is just `#define`d to `calloc`. We could grep `libmbedcrypto.a` for `calloc`, or arrange to define `mbedtls_calloc` to `if_you_get_a_link_error_this_test_is_failing`, or any other solution that results in the component failing if the core calls `mbedtls_calloc`.

Reply:
> or arrange to define mbedtls_calloc to if_you_get_a_link_error_this_test_is_failing

I found this to be the simplest solution. Unfortunately test component is still failing in https://github.com/Mbed-TLS/mbedtls/pull/9302/commits/079ad6c883cf072163f471d87e4fe9b97048c04b, but I'm working on that to fix it.

Reply:
As of https://github.com/Mbed-TLS/mbedtls/pull/9302/commits/c22f65211e8b6c686a71de41833243ba66c2b78f this PR should have the CI green and therefore it should be ready for review ;)